### PR TITLE
Deneb Support

### PR DESCRIPTION
**Pull request by olegshmuelov**
Integration of Deneb Fork Compatibility Features

Changes:
- Fixed HD path validation function. Unrelated to Deneb, but required to run the real block proposal test.
- Replaced phase0.Slot with uint64 due to encoding changes in go-eth2-client (they changed encoding from number to string, so we use uint64 to preserve old behavior)
- Added proposal tests based on a real block from Prater

### CODE REVIEW

**Review by y0sher — APPROVED** [review]
is this everyhing we need?

**Review by y0sher — APPROVED** [review]

**Review by moshe-blox — APPROVED** [review]

**Review by y0sher — APPROVED** [review]

File: core/master_derivable_key.go [hunk 1]
```diff
@@ -88,6 +88,6 @@ func (master *MasterDerivableKey) Derive(relativePath string) (*HDKey, error) {
 }
 
 func validateRelativePath(relativePath string) bool {
-	match, _ := regexp.MatchString(`^(\/(\d\d?\d?))+$`, relativePath)
+	match, _ := regexp.MatchString(`^\/\d+(\/0)(\/0)?$`, relativePath)
```
Comment (moshe-blox) [inline core/master_derivable_key.go:91]: Why did this change?

**Review by moshe-blox — CHANGES_REQUESTED** [review]
LGTM, gj 🔥

Left just one question.

File: core/master_derivable_key.go (hunk 1 above)
Comment (olegshmuelov) [reply core/master_derivable_key.go]: reverted to support more then 3 digits derivation  paths

**Review by moshe-blox — APPROVED** [review]

**Review by y0sher — APPROVED** [review]
looks good gj

File: signer/sign_beacon_block_test.go [hunk 2]
```diff
@@ -195,6 +196,63 @@ func TestBenchmarkBlockProposalCapella(t *testing.T) {
...
...48279283be1ecbdbaac58f8ca0627d8b87a4442835357763bc5971dd321c87540584a10711b38e894cc45ed55d6741c978e026e6d0e5e665c88d792216f236c0658de098e3c084a458a79b037a0888a58a6a659f6b8b367e07cba21b00109de651505c0cba21790b408bac425a2d9179f6a263d0e28b43fee135dd2196ca6bd848a909230f1e9087b15a069adaaf6962c94c6608c1e17fbfe0a66108e2232cb315263821c88c0f4c4759c3bf62adad5a249d2f80484470bebfdade20274f3fb77135e373190c67aba6d3d2a9bcdd6aa42a345750748d4f9b9779db41b30f52cfd23c5af0a19b40188a30a03b3afb557150a3d293da4222fcfa61456e95187d35605bae235c88eae336963cdac4c38729fc17707991b753e4c62d9d78abcf73aa533554b52e7aa1fbe00229d3f56a0b3e1827326eeaed644d776e0d011d9adbd69adbcd83600458ac23743c6b423105faeb0b658e3214bcd4231b19ae8f2570873839986ded21020a2b8e306a0b8a71d3c88c48a812149383dd1f71a99d6d5a2b19078e1b0a4530f9a59a5ceb73aeb29f7629657ed101b8bbeef8c24567933daec8d57ed379a63711fc0f4aacdf2db513ef0439811fa0d66b2be6166eda85b3d92ec0293a66544021ebb15095e7ff8e17b3133204af2d3c5878b691f077f859981ff5e41d769da367fc60c09bcfaef6286291985f94d65a257881b07ecf26923d296142cb6cde7fa5d53d048f9d8cc073eea785e26c0419ec80977e226647ae4f454f9b5ca8735ffe4988320cea8f0129eb7bede341923aa78e010faed31a6235fdeb32d536687dda933cc70d843f4e19810a4084b2f95c3c
```
[the remark this answers is not in the document]
Comment (olegshmuelov) [reply signer/sign_beacon_block_test.go:251]: yeah, the purpose of this test is to test real block and compare the signatures

File: core/master_derivable_key.go [hunk 3]
```diff
@@ -88,6 +88,6 @@ func (master *MasterDerivableKey) Derive(relativePath string) (*HDKey, error) {
 }
 
 func validateRelativePath(relativePath string) bool {
-	match, _ := regexp.MatchString(`^(\/(\d\d?\d?))+$`, relativePath)
```
Comment (alonmuroch) [inline core/master_derivable_key.go:91]: why is this changed?

File: signer/sign_beacon_block_test.go (hunk 2 above)
Comment (alonmuroch) [inline signer/sign_beacon_block_test.go:211]: where is this taken from?

File: signer/sign_blinded_beacon_block_test.go [hunk 4]
```diff
@@ -107,6 +109,78 @@ func TestBenchmarkBlindedBlockProposalCapella(t *testing.T) {
...
...004d0010000820495202200018002a880040a9081849d2040192","prev_randao":"0xc5647b9f7b3ca7b07b86365095fe47f84ca193c009d91182f9d254b2e9bd59ae","block_number":"10389257","gas_limit":"30000000","gas_used":"8753279","timestamp":"1705494252","extra_data":"0xd883010d0a846765746888676f312e32312e36856c696e7578","base_fee_per_gas":"9","block_hash":"0x202d5d304bf17fec211cd13186fb766dbe610c12574d4c9eadd7e6fdae9ee050","transactions_root":"0x4e2efc021603f6106e875c8d74d5af83103fda595bd5b6cff0f0a4a8935a03d2","withdrawals_root":"0x2457c13873dcd199ffdc4a51f4ae9812d7409418c8e015a7557d8ebceb110fe7","blob_gas_used":"655360","excess_blob_gas":"86507520"},"bls_to_execution_changes":[],"blob_kzg_commitments":["0xc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","0xb24758fbdea9852ae96086be16e5634c18a51acc31c9b2b97f7e330bb41065987f73376073c3e1d28f11ccb2d2fc03c2","0xa45a58d23333471a49032949ac925593f2958977c000b0da114092e79910981550fbca5f57c9d9c9aa1e8a90ac0232cf","0x852b25e16fef449d320869f8984f6bda8a04f15a9e6d6cb0fc35c16c8c9e03c377bc387bbca75af799d739cd229672af","0x99943f758bb009192d7dce3d6172e9d466367ac717c6ecab30bd5cb98badf8c1290a8a0529d56f3812f97504def2d970"]}}`)
```
Comment (alonmuroch) [inline signer/sign_blinded_beacon_block_test.go:124]: where is this taken from?

File: stores/inmemory/store.go [hunk 5]
```diff
@@ -26,7 +26,7 @@ type InMemStore struct {
 	highestAttestation     map[string]*phase0.AttestationData
 
 	highestProposalLock sync.RWMutex
-	highestProposal     map[string]phase0.Slot
+	highestProposal     map[string]uint64
```
Comment (alonmuroch) [inline stores/inmemory/store.go:29]: why is this changed?

File: core/master_derivable_key.go (hunk 3 above)
Comment (olegshmuelov) [reply core/master_derivable_key.go:91]: The primary reason for updating the regular expression in `validateRelativePath` was to facilitate testing of a real block that was signed by a validator whose keys we possess, which allowed us to perform more thorough and realistic testing scenario. Specifically, the issue we faced was related to our inability to derive the validator's key with the relativePath `/3536/0/0`

btw more tests were added due to this change [here](https://github.com/bloxapp/eth2-key-manager/blob/9ccffcf86f2a585f6bc09a90f49b8f6bc4220bb3/core/hd_key_test.go)

File: signer/sign_beacon_block_test.go (hunk 2 above)
Comment (olegshmuelov) [reply signer/sign_beacon_block_test.go:211]: there is a comment above the func testing this block

File: signer/sign_blinded_beacon_block_test.go (hunk 4 above)
Comment (olegshmuelov) [reply signer/sign_blinded_beacon_block_test.go:124]: there is a comment above the func testing this block

File: stores/inmemory/store.go (hunk 5 above)
Comment (olegshmuelov) [reply stores/inmemory/store.go:29]: > as written in description of the PR

Replaced phase0.Slot with uint64 due to encoding changes in go-eth2-client (they changed encoding from number to string, so we use uint64 to preserve old behavior)
